addon/mixins/route.js

```diff
@@ -195,13 +95,21 @@ const RouteMixin = Mixin.create({
     @method infinityModel
     @param {String} modelName The name of the model.
     @param {Object} options Optional, the perPage and startingPage to load from.
-    @param {Object} boundParams Optional, any route properties to be included as additional params.
     @return {Ember.RSVP.Promise}
   */
   infinityModel(modelName, options, boundParams) {
-    options = options ? objectAssign({}, options) : {};
+    if (modelName === undefined) {
+      throw new Ember.Error("Ember Infinity: You must pass a Model Name to infinityModel");
+    }
+    if (boundParams) {
+      throw new Ember.Error("Bound params are now deprecated. Please pass explicitly as second param to infinityModel");
```

Comment:
Can we do an Ember Deprecate here?  Or if you'd rather a hard error, can we prepend something like `Ember Infinity` so the developer at least knows where the error is from

We should also potentially link to an anchor in the README.md for moving off of bound params.

Do you know what the actual upgrade path is here?  We may actually not be able to provide an upgrade path without allowing developers to extend the infinity model.